### PR TITLE
Hackerrank's 4th set of exercises with log4j2 and attendance - Brandon Beltran

### PR DESCRIPTION
**Pull request by Armastardo**
I think I configured the log properly! At least I could write to a logfile with no issues. :)

### CODE REVIEW

**Comment by Armastardo** [comment]
Revisando master me di cuenta que por el camino me volaron los ejercicios que subí en https://github.com/globant-bootcamp/bootcamp-java-august-2021/pull/50 , los he vuelto a subir (por si ven que hay archivos de más). La carpeta de estos últimos ejercicios es la 04.

File: BrandonBeltran/hackerrank/03/JavaArrayList.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        int collections = scanner.nextInt();
+        List <List> numbers = new ArrayList<>();
+
+        for(int collectionIterator = 0; collectionIterator < collections; collectionIterator++){
+            int size = scanner.nextInt();
+
+            ArrayList <Integer> numberSubList = new ArrayList<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaArrayList.java:17]: Code for interface not for implementation

File: BrandonBeltran/hackerrank/03/JavaArrayList.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class JavaArrayList {
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        int collections = scanner.nextInt();
+        List <List> numbers = new ArrayList<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaArrayList.java:12]: The best practice in Collections or Map variable name, is use name in singular and them the interface name: numberList, change it in all the variables

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatter.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package bootcamp;
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatter.java:3]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package bootcamp;
+
+import java.util.List;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:8]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    private static final String INDIA_CURRENCY_SIGN = "Rs.";
+    private static final String CHINA_CURRENCY_SIGN = "￥";
+    private static final String FRANCE_CURRENCY_SIGN = "€";
+
+    public static void main(String[] args) {
+
+        /* This solution was made before I realized NumberFormat.getCurrencyInstance() existed.
+        * It's preserved here just for completion purposes */
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:22]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        Scanner scanner = new Scanner(System.in);
+        double payment = scanner.nextDouble();
+        scanner.close();
+
+        String [] paymentString = String.valueOf(payment).split("\\.", 0);
+        String decimals = paymentString[1].substring(0,2);
+        String numerals = paymentString[0];
+
+        List<String> splittedString = splitEveryN(numerals, 3);
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:31]: Best practice in Collections or Map Variable Name, is to use singular name and them interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+        String commaSeparated = splittedString.stream().map(Object::toString).
+                collect(Collectors.joining(",")).toString();
+        String spaceSeparated = commaSeparated.replaceAll(",", " ");
+
+        String us = US_CURRENCY_SIGN+commaSeparated+"."+decimals;
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:37]: Separate the + by a space
Why do you do it in some places and not others? Homologous your code, change it in all the code

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        String india = INDIA_CURRENCY_SIGN+commaSeparated+"."+decimals;
+        String china = CHINA_CURRENCY_SIGN+commaSeparated+"."+decimals;
+        String france = commaSeparated+","+decimals+" "+FRANCE_CURRENCY_SIGN;
+
+        System.out.println("US: " + us);
+        System.out.println("India: " + india);
+        System.out.println("China: " + china);
+        System.out.println("France: " + france);
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:47]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:48]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    public static List<String> splitEveryN(String string, int nLength){
+        List<String> splittedString = new ArrayList<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:52]: Best practice in Collections or Map Variable Name, is to use singular name and them interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+
+        int positionSpliter = 0;
+        if((string.length()%nLength) != 0){
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:55]: Separate the % by a space

Comment (globant-bootcamp) [reply BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:55]: And in all the code the same

Comment (globant-bootcamp) [reply BrandonBeltran/hackerrank/03/JavaCurrencyFormatterLegacy.java:55]: And in all the code the same

File: BrandonBeltran/hackerrank/03/JavaHashset.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+/* This solution is the one I came on my own without Hackerrank's code.
+*  To be honest, I like it better, I include it for the sake of completion*/
+
+public class JavaHashset {
+    public static void main(String[] args) {
+
+        Scanner scanner = new Scanner(System.in);
+        int sets = Integer.parseInt(scanner.nextLine());
+
+        Set<String> names = new HashSet();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaHashset.java:16]: Best practice in Collections or Map Variable Name, is to use singular name and them interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/JavaList.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Iterator;
+import java.util.List;
+import java.util.Scanner;
+
+public class JavaList {
+
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+        int size = scanner.nextInt();
+        List <Integer> numbers = new ArrayList<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaList.java:13]: Best practice in Collections or Map Variable Name, is to use singular name and them interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/JavaMap.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class JavaMap {
+
+    public static void main(String[] args) {
+
+        Map<String, Integer> phoneBook = new HashMap<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/JavaMap.java:11]: Best practice in Collections or Map Variable Name, is to use singular name and interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/RandomNoteSolve.java [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RandomNoteSolve {
+
+    public static void checkMagazine(List<String> magazine, List<String> note) {
+
+        Map<String, Integer> magazineWords = wordsToHashMap(magazine);
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/RandomNoteSolve.java:11]: Best practice in Collections or Map Variable Name, is to use singular name and interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/RandomNoteSolve.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+        Map<String, Integer> noteWords = wordsToHashMap(note);
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/RandomNoteSolve.java:12]: same

File: BrandonBeltran/hackerrank/03/RandomNoteSolve.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+                doable = "No";
+                break;
+            }
+        }
+
+        System.out.println(doable);
+    }
+
+    public static HashMap<String, Integer> wordsToHashMap(List<String> wordsList){
+        HashMap<String, Integer> convertedList = new HashMap<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/RandomNoteSolve.java:29]: Code for interface not for implementation

File: BrandonBeltran/hackerrank/03/RansomNote.java [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+    public static void main(String[] args) throws IOException {
+
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
+
+        String[] totalWords = bufferedReader.readLine().replaceAll(WHITESPACE_AND_EOL, "").split(" ");
+
+        int magazineTotalWords = Integer.parseInt(totalWords[0]);
+        int noteTotalWords = Integer.parseInt(totalWords[1]);
+
+        List<String> magazine = Stream.of(bufferedReader.readLine().replaceAll(WHITESPACE_AND_EOL, "").split(" "))
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/RansomNote.java:23]: Best practice in Collections or Map Variable Name, is to use singular name and them interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/03/RansomNote.java [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+                .collect(toList());
+
+        List<String> note = Stream.of(bufferedReader.readLine().replaceAll(WHITESPACE_AND_EOL, "").split(" "))
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/RansomNote.java:26]: same

File: BrandonBeltran/hackerrank/03/Student.java [hunk 20]
```diff
@@ -0,0 +1,34 @@
+package bootcamp;
+
+class Student implements Comparable<Student>{
+    private int id;
+    private String fname;
+    private double cgpa;
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/Student.java:6]: cgpa? improve the variable name

File: BrandonBeltran/hackerrank/03/Student.java [hunk 21]
```diff
@@ -0,0 +1,34 @@
+package bootcamp;
+
+class Student implements Comparable<Student>{
+    private int id;
+    private String fname;
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/03/Student.java:5]: fname? improve the variable name

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+
+        Scanner scanner = new Scanner(System.in);
+        double payment = scanner.nextDouble();
+        scanner.close();
+
+        String [] paymentString = String.valueOf(payment).split("\\.", 0);
+        String decimals = paymentString[1].substring(0,2);
+        String numerals = paymentString[0];
+
+        List<String> splittedString = splitEveryN(numerals, 3);
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:37]: Best practice in Collections or Map Variable Name, is to use singular name and interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 23]
```diff
@@ -0,0 +1,84 @@
...
+        String us = US_CURRENCY_SIGN+commaSeparated+"."+decimals;
+        String india = INDIA_CURRENCY_SIGN+commaSeparated+"."+decimals;
+        String china = CHINA_CURRENCY_SIGN+commaSeparated+"."+decimals;
+        String france = commaSeparated+","+decimals+" "+FRANCE_CURRENCY_SIGN;
+
+        logger.info("US: " + us);
+        logger.info("India: " + india);
+        logger.info("China: " + china);
+        logger.info("France: " + france);
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:52]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 24]
```diff
@@ -0,0 +1,84 @@
...
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:53]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 25]
```diff
@@ -0,0 +1,84 @@
...
+    private static final String INDIA_CURRENCY_SIGN = "Rs.";
+    private static final String CHINA_CURRENCY_SIGN = "￥";
+    private static final String FRANCE_CURRENCY_SIGN = "€";
+
+    public static void main(String[] args) {
+
+        /* This solution was made before I realized NumberFormat.getCurrencyInstance() existed.
+         * It's preserved here just for completion purposes */
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:28]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 26]
```diff
@@ -0,0 +1,84 @@
...
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:13]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatter.java [hunk 27]
```diff
@@ -0,0 +1,23 @@
+package bootcamp;
+
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatter.java:3]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 28]
```diff
@@ -0,0 +1,84 @@
...
+        logger.info("India: " + india);
+        logger.info("China: " + china);
+        logger.info("France: " + france);
+
+
+
+    }
+
+    public static List<String> splitEveryN(String string, int nLength){
+        List<String> splittedString = new ArrayList<>();
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:58]: Best practice in Collections or Map Variable Name, is to use singular name and interface name: numberList, change it in all variables

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java [hunk 29]
```diff
@@ -0,0 +1,84 @@
...
+
+        int positionSpliter = 0;
+        if((string.length()%nLength) != 0){
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatterLegacy.java:61]: separate the % by a space

File: BrandonBeltran/hackerrank/04/JavaStaticInitializerBlock.java [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+    }
+
+    public static void main(String[] args){
+        if(positiveNumbersFlag){
+            int area = breadth * height;
+            logger.info(area);
+        }
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaStaticInitializerBlock.java:33]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaStaticInitializerBlock.java [hunk 31]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaStaticInitializerBlock.java:35]: delete unnecessary blank space

File: BrandonBeltran/hackerrank/04/JavaCurrencyFormatter.java [hunk 32]
```diff
@@ -0,0 +1,23 @@
...
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.Scanner;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+public class JavaCurrencyFormatter {
+    private static final Logger logger = LogManager.getLogger(Test.class);
```
Comment (globant-bootcamp) [inline BrandonBeltran/hackerrank/04/JavaCurrencyFormatter.java:11]: If this variable is static final, what is the correct name? change in all the code

**Review by globant-bootcamp — CHANGES_REQUESTED** [review]

**Review by globant-bootcamp — APPROVED** [review]

**Pull request merged**